agr_literature_service/api/models/mod_model.py

```diff
@@ -43,13 +43,16 @@ class ModModel(Base, AuditedModel):
         nullable=False
     )
 
-    taxon_ids = Column(
+    taxon_ids: Column = Column(
         ARRAY(String()),
         unique=False,
         nullable=True
     )
 
-    referencetypes = relationship("ModReferencetypeAssociationModel")
+    referencetypes = relationship(
+        "ModReferencetypeAssociationModel",
+        back_populates="mod"
```

Comment:
This is actually needed now, and it wasn't needed before ?  Just making sure because sometimes extra back_populates were making things small, and @valearna removed them

Reply:
I added the back_populates="mod" in the referencetypes relationship to establish
a bi-directional relationship between the ModModel and ModReferencetypeAssociationModel classes.
This allows SQLAlchemy to automatically manage the relationship in both directions.

So when we access mod.referencetypes (on ModModel), it fetches the related
ModReferencetypeAssociationModel instances associated with that mod.
When we access referencetype.mod (on ModReferencetypeAssociationModel), it fetches the
associated ModModel instance.

Without adding this, all programs and modules using mod generate a BIG warning. Remember we
have talked about this in slack channel. And I asked ChatGPT for this solution.

Reply:
Ah, okay.  I saw that you and Valerio were discussing it, but I wasn't clear.